instrumentation/sinatra/lib/opentelemetry/instrumentation/sinatra/extensions/tracer_extension.rb

```diff
@@ -13,23 +13,25 @@ module Extensions
         # Sinatra extension that installs TracerMiddleware and provides
         # tracing for template rendering
         module TracerExtension
-          # Sinatra hook after extension is registered
-          def self.registered(app)
-            # Create tracing `render` method
-            ::Sinatra::Base.module_eval do
-              def render(_engine, data, *)
-                template_name = data.is_a?(Symbol) ? data : :literal
+          # Contants patches for `render` method
+          module RenderPatches
+            def render(_engine, data, *)
+              template_name = data.is_a?(Symbol) ? data : :literal
 
-                Sinatra::Instrumentation.instance.tracer.in_span(
-                  'sinatra.render_template',
-                  attributes: { 'sinatra.template_name' => template_name.to_s }
-                ) do
-                  super
-                end
+              Sinatra::Instrumentation.instance.tracer.in_span(
+                'sinatra.render_template',
+                attributes: { 'sinatra.template_name' => template_name.to_s }
+              ) do
+                super
               end
             end
-            app.use(*OpenTelemetry::Instrumentation::Rack::Instrumentation.instance.middleware_args)
-            app.use(Middlewares::TracerMiddleware)
+          end
+
+          # Sinatra hook after extension is registered
+          def self.registered(app)
+            # Create tracing `render` method
+            ::Sinatra::Base.prepend(RenderPatches)
```

Comment:
I appreciate the use of a `module` we can `prepend` rather than `module_eval`. The later always causes me a little anxiety, and to go re-read the docs; probably trauma from over-use and being overly-clever back in the Ruby meta-programming heydays. 😆